packages/story-editor/src/app/media/utils/useProcessMedia.js

```diff
@@ -478,11 +479,95 @@ function useProcessMedia({
     ]
   );
 
+  /**
+   * Crop video using FFmpeg.
+   *
+   * @param {import('@googleforcreators/media').Resource} resource Resource object.
+   * @param {Object<{newWidth: number, newHeight: number, cropElement: Element}>} cropParams Crop params.
+   */
+  const cropExistingVideo = useCallback(
+    ({ id: elementId, resource: oldResource }, cropParams) => {
+      const { id: resourceId, src: url, mimeType, isOptimized } = oldResource;
+      const { newWidth, newHeight, cropElement } = cropParams;
+
+      const onUploadSuccess = ({ id, resource }) => {
+        copyResourceData({ oldResource, resource });
+        updateOldTranscodedObject(oldResource.id, resource.id, 'source-video');
+        updateElementById({
+          elementId,
+          properties: {
+            x: cropElement.x < 0 ? 0 : cropElement.x,
+            y:
+              cropElement.y < DANGER_ZONE_HEIGHT
+                ? -DANGER_ZONE_HEIGHT
+                : cropElement.y,
+            width: newWidth,
+            height: newHeight,
+            resource: {
+              ...resource,
+              id: oldResource.id,
+              poster: null,
+              posterId: 0,
+            },
+          },
+        });
+
+        // onUploadSuccess is also called with previousResourceId,
+        // for which we don't need to run this.
+        if (id === resource.id) {
+          postProcessingResource(resource);
+        }
+      };
+
+      const onUploadProgress = ({ resource }) => {
+        const newResourceWithCanvasId = {
+          ...resource,
+          id: cropElement.resource.id,
+        };
+        updateExistingElementById(elementId, {
+          ...newResourceWithCanvasId,
+        });
+      };
+
+      const process = async () => {
+        let file = false;
+        try {
+          file = await fetchRemoteFile(url, mimeType);
+          await uploadMedia([file], {
+            onUploadSuccess,
+            onUploadProgress,
+            cropVideo: true,
+            additionalData: {
+              original_id: resourceId,
+              cropParams,
+              mediaSource: isOptimized ? 'video-optimization' : 'editor',
+            },
+            originalResourceId: resourceId,
```

Comment:
```suggestion
            originalResourceId: resourceId,
             resource: {
              ...resource,
              poster: null,
              posterId: 0,
            },
```

I think we need to the poster image workaround here and maybe even the new height and width.